homeassistant/components/vizio/media_player.py

```diff
@@ -32,35 +45,56 @@
 MIN_TIME_BETWEEN_FORCED_SCANS = timedelta(seconds=1)
 MIN_TIME_BETWEEN_SCANS = timedelta(seconds=10)
 
-SUPPORTED_COMMANDS = SUPPORT_TURN_ON | SUPPORT_TURN_OFF \
-                     | SUPPORT_SELECT_SOURCE \
-                     | SUPPORT_NEXT_TRACK | SUPPORT_PREVIOUS_TRACK \
-                     | SUPPORT_VOLUME_MUTE | SUPPORT_VOLUME_STEP \
-                     | SUPPORT_VOLUME_SET
+COMMON_SUPPORTED_COMMANDS = (
+    SUPPORT_SELECT_SOURCE |
+    SUPPORT_TURN_ON |
+    SUPPORT_TURN_OFF |
+    SUPPORT_VOLUME_MUTE |
+    SUPPORT_VOLUME_SET |
+    SUPPORT_VOLUME_STEP
+)
+
+SUPPORTED_COMMANDS = {
+    'soundbar': COMMON_SUPPORTED_COMMANDS,
+    'tv': (
+        COMMON_SUPPORTED_COMMANDS |
+        SUPPORT_NEXT_TRACK |
+        SUPPORT_PREVIOUS_TRACK
+    )
+}
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
     vol.Required(CONF_HOST): cv.string,
-    vol.Required(CONF_ACCESS_TOKEN): cv.string,
+    vol.Optional(CONF_ACCESS_TOKEN): cv.string,
     vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
     vol.Optional(CONF_SUPPRESS_WARNING, default=False): cv.boolean,
+    vol.Optional(CONF_DEVICE_CLASS, default=DEFAULT_DEVICE_CLASS):
+        vol.All(cv.string, vol.Lower, vol.In(['tv', 'soundbar'])),
     vol.Optional(CONF_VOLUME_STEP, default=DEFAULT_VOLUME_STEP):
         vol.All(vol.Coerce(int), vol.Range(min=1, max=10)),
 })
 
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up the VizioTV media player platform."""
+    """Set up the Vizio media player platform."""
     host = config.get(CONF_HOST)
     token = config.get(CONF_ACCESS_TOKEN)
     name = config.get(CONF_NAME)
     volume_step = config.get(CONF_VOLUME_STEP)
-
-    device = VizioDevice(host, token, name, volume_step)
+    device_type = config.get(CONF_DEVICE_CLASS)
+    device = VizioDevice(host, token, name, volume_step, device_type)
     if device.validate_setup() is False:
-        _LOGGER.error("Failed to set up Vizio TV platform, "
+        _LOGGER.error("Failed to set up Vizio platform, "
                       "please check if host and API key are correct")
         return
 
+    if (token is None or token == "") and device_type == "tv":
```

Comment:
We should catch this with a custom config validator.

Reply:
Per https://github.com/home-assistant/home-assistant/pull/22294#discussion_r276069246 I will consider this resolved. I will keep the other thread open in case a solution can be identified

Reply:
See my comment above.

Reply:
Custom validator added per above comment